main.go

```diff
@@ -791,8 +842,8 @@ func parseCustomDNSLinkResolvers(customDNSResolvers []string) (madns.BasicResolv
 			return nil, fmt.Errorf("invalid DNS resolver: %s", s)
 		}
 		domain := strings.TrimSpace(split[0])
-		resolverUrl := strings.TrimSpace(split[1])
-		customDNSResolverMap[domain] = resolverUrl
+		resolverURL := strings.TrimSpace(split[1])
+		customDNSResolverMap[domain] = resolverURL
```

Comment:
Maybe worth parsing the URL
```suggestion
		resolverURL, err := url.Parse(strings.TrimSpace(split[1]))
		if err != nil {
			return nil, err
		}
		customDNSResolverMap[domain] = resolverURL.String()
```